src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -245,6 +245,18 @@ CTranslatorDXLToPlStmt::GetPlannedStmtFromDXL(const CDXLNode *dxlnode,
 
 	topslice = &planned_stmt->slices[0];
 
+	// Always set the REWIND flag for subplans, since ORCA cannot generate
+	// InitPlans.
+	List *subplan_list =
+		gpdb::ExtractNodesPlan(planned_stmt->planTree, T_SubPlan, true);
```

Comment:
I noticed that I missed one unclear aspect of ExtractNodesPlan in original patch. The case is that `extract_nodes_plan` [does not walk into SubPlan subquery](https://github.com/arenadata/gpdb/blob/e8aa067ffdaa4618d8483997ecf341c6d1ea0969/src/backend/optimizer/util/walkers.c#L701-L707), and it leads to incomplete subplan processing in original and this patch, when one SubPlan contains another one inside. Like

<details>
  <summary>Ex</summary>

  ```

create table t1 (i int, j int);
create table t2 (i int, j int);
create table t3 (i int, j int) distributed randomly;
explain select * from t1 where t1.i > ALL (select t2.j from t2 where t2.i=t1.i and t2.j > ALL (select i from t3 where t3.i=t2.i));
Result  (cost=0.00..1356691736.76 rows=1 width=8)
   Filter: (SubPlan 2)
   ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..431.00 rows=1 width=8)
         ->  Seq Scan on t1  (cost=0.00..431.00 rows=1 width=8)
   SubPlan 2
     ->  Result  (cost=0.00..1324032.27 rows=1 width=1)
           Filter: ((CASE WHEN (sum((CASE WHEN (t1.i <= t2.j) THEN 1 ELSE 0 END)) IS NULL) THEN true WHEN (sum((CASE WHEN (t2.j IS NULL) THEN 1 ELSE 0 END)) > '0'::bigint) THEN NULL::boolean
 WHEN (t1.i IS NULL) THEN NULL::boolean WHEN (sum((CASE WHEN (t1.i <= t2.j) THEN 1 ELSE 0 END)) = '0'::bigint) THEN true ELSE false END) = true)
           ->  Aggregate  (cost=0.00..1324032.27 rows=1 width=16)
                 ->  Result  (cost=0.00..1324032.27 rows=1 width=4)
                       Filter: (t2.i = t1.i)
                       ->  Materialize  (cost=0.00..1324032.27 rows=1 width=8)
                             ->  Gather Motion 3:1  (slice2; segments: 3)  (cost=0.00..1324032.27 rows=1 width=8)
                                   ->  Seq Scan on t2  (cost=0.00..1324032.27 rows=1 width=8)
                                         Filter: (SubPlan 1)
                                         SubPlan 1
                                           ->  Result  (cost=0.00..431.00 rows=1 width=1)
                                                 Filter: ((CASE WHEN (sum((CASE WHEN (t2.j <= t3.i) THEN 1 ELSE 0 END)) IS NULL) THEN true WHEN (sum((CASE WHEN (t3.i IS NULL) THEN 1 ELSE 0 E
ND)) > '0'::bigint) THEN NULL::boolean WHEN (t2.j IS NULL) THEN NULL::boolean WHEN (sum((CASE WHEN (t2.j <= t3.i) THEN 1 ELSE 0 END)) = '0'::bigint) THEN true ELSE false END) = true)
                                                 ->  Aggregate  (cost=0.00..431.00 rows=1 width=16)
                                                       ->  Result  (cost=0.00..431.00 rows=1 width=4)
                                                             Filter: (t3.i = t2.i)
                                                             ->  Materialize  (cost=0.00..431.00 rows=1 width=4)
                                                                   ->  Broadcast Motion 3:3  (slice3; segments: 3)  (cost=0.00..431.00 rows=1 width=4)
                                                                         ->  Seq Scan on t3  (cost=0.00..431.00 rows=1 width=4)
 Optimizer: GPORCA
(24 rows)

```
</details>

Therefore, exectutor does not initialize nested subplan with rewind flag.

Probably we should rework approach on ORCA side here and create ticket for backporting it to 6.x